onnxruntime/core/providers/mkldnn/activation/activations.cc

```diff
@@ -0,0 +1,238 @@
+/* Copyright(C) 2018 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions
+and limitations under the License.
+==============================================================================*/
+
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/activation/activations.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+namespace {
+// Struct which encapsulates parameters for MKLDNN Pool primitive.
+struct ReluParams {
+  mkldnn::memory::dims& src_dims;
+  mkldnn::memory::dims& dst_dims;
+  size_t num_dimensions;
+
+  ReluParams(mkldnn::memory::dims& src_dims, mkldnn::memory::dims& dst_dims, 
+		size_t dimensions = 0)
+      : src_dims(src_dims),
+        dst_dims(dst_dims),
+        num_dimensions(dimensions) {}
+
+  // Used as the key for Pool Primitive Reuse Pool.
+  std::string ToString() const {
+    std::string key;
+    key.reserve(128);
```

Comment:
probably don't need 128 bytes for just a src and dst dims.

Reply:
will change this to 64 bytes